### ISSUE TITLE
Blazorise components don't work with CSS Isolation

### ISSUE THREAD

**Issue by ghost** [stsrki/Blazorise#1599]
Not sure if this is something you can fix or not, but Blazorise components don't get scope attributes added so CSS Isolation doesn't work.

* Create a page, e.g. `Test.razor`, add a Blazorise component and a plain HTML element.
* Add a `Test.razor.css` file, and reference `<Root namespace>.styles.css` in your index.html as described [here](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/css-isolation?view=aspnetcore-5.0)
* Add a class to the .css file with `background-color: red;` or something else obvious, add the class to your blazorise component and the html element.
* Run the project and open the page
* The class has not been applied to the component, only the element.

**Comment by stsrki** [comment]
I tried it and you're right, it's not working. The reason I suspect is that Blazor does not generate a special attribute with a random ID for custom components.

![image](https://user-images.githubusercontent.com/900302/101880948-0204ce00-3b94-11eb-916f-6df28b393141.png)

I will try to investigate more but for now, it's low on a priority list. Maybe for the next mileston.

**Comment by gerfen** [comment]
If you wrap you pages/components in a <div/> and use deep:: in your css/scss, then css isolation mostly works.

**Comment by WAELKASSEM** [comment]
> 
> 
> If you wrap your pages/components in a `<div/>` and use deep:: in your css/scss, then css isolation mostly works.

Here's how you do it.

```
::deep .myclass {
    color: red;
}
```
```
<div>
    <Button Class="myclass">abc</Button>
</div>
```
**The Div is very Important**

**Comment by Jinjinov** [comment]
> I tried it and you're right, it's not working. The reason I suspect is that Blazor does not generate a special attribute with a random ID for custom components.
> 
> ![image](https://user-images.githubusercontent.com/900302/101880948-0204ce00-3b94-11eb-916f-6df28b393141.png)
> 
> I will try to investigate more but for now, it's low on a priority list. Maybe for the next mileston.

What is the `_bl_10` in the Blazorise button? I have seen this on all my Blazorise buttons.

Is this something that the Blazorise library generates?

My first thought was that `b-z90skj8z01` was replaced with `_bl_10` by the Blazorise library... or by some other code... just guessing here...

**Comment by stsrki** [comment]
It is what gets generated by the blazor when you create ElementReference:

`builder.AddElementReferenceCapture( ... );`

**Comment by stsrki** [comment]
https://docs.telerik.com/blazor-ui/knowledge-base/common-css-isolation

**Comment by stsrki** [comment]
There is currently no way of doing this feature. Closing until Blazor gets proper support for custom components CSS isolation. If ever. Once it is done I will create a new ticket for sure.

@WAELKASSEM workaround is the best option at this time.

**Comment by KroMignon** [comment]
> @WAELKASSEM workaround is the best option at this time.

This not a workaround but the way CSS isolation works for children components
=> cf. Microsoft documentation https://docs.microsoft.com/en-us/aspnet/core/blazor/components/css-isolation?view=aspnetcore-5.0
